src/sas/qtgui/Utilities/DocViewWidget.py

```diff
@@ -0,0 +1,216 @@
+import sys
+import os
+from typing import Optional
+import logging
+
+
+from PySide6 import QtGui, QtWebEngineWidgets, QtCore, QtWidgets, QtWebEngineCore
+from twisted.internet import threads
+
+from .UI.DocViewWidgetUI import Ui_DocViewerWindow
+from sas.qtgui.Utilities.TabbedModelEditor import TabbedModelEditor
+from sas.qtgui.Utilities import GuiUtils
+
+
+class DocViewWindow(QtWidgets.QDialog, Ui_DocViewerWindow):
+    """
+    Instantiates a window to view documentation using a QWebEngineViewer widget
+    """
+
+    def __init__(self, parent=None, source=None):
+        super(DocViewWindow, self).__init__(parent._parent)
+        self.parent = parent
+        self.setupUi(self)
+        # Disable the context help icon
+        self.setWindowFlags(self.windowFlags() & ~QtCore.Qt.WindowContextHelpButtonHint)
+        self.setWindowTitle("Documentation Viewer")
+
+        # Necessary globals
+        self.source = source
+        self.regen_in_progress = False
+
+        self.initializeSignals() # Connect signals
+
+        self.regenerateIfNeeded()
+
+    def initializeSignals(self):
+        """
+        Initialize Signals 
+        """
+
+        self.editButton.clicked.connect(self.onEdit)
+        self.closeButton.clicked.connect(self.onClose)
+        self.parent.communicate.documentationRegeneratedSignal.connect(self.refresh)
+
+    def onEdit(self):
+        """
+        Open editor (TabbedModelEditor) window.
+        """
+        # Convert QUrl to pathname:
+        from re import findall, sub
+        # Extract path from QUrl
+        path = findall(r"(?<=file:\/\/\/).+\.html", str(self.webEngineViewer.url()))
+        # Test to see if we're dealing with a model html file or other html file
+        if "models" in path[0]:
+            py_base_file_name = os.path.splitext(os.path.basename(path[0]))[0]
+            self.editorWindow =  TabbedModelEditor(parent=self.parent,
+                                                   edit_only=True,
+                                                   load_file=py_base_file_name,
+                                                   model=True)
+        else:
+            # Remove everything before /user/ (or \user\)
+            file = sub(r"^.*?(?=[\/\\]user)", "", path[0])
+
+            # index.html is the only rst file outside of /user/ folder-- set it manually
+            if "index.html" in file:
+                file = "/index.html"
+            self.editorWindow =  TabbedModelEditor(parent=self.parent,
+                                                   edit_only=True,
+                                                   load_file=file,
+                                                   model=False)
```

Comment:
Separate the path manipulation logic from the editor startup. You can set `file=` in each branch of the condition, then use `load_file=file` after the condition to start the editor.

I don't think you need to escape the forward slashes in your regular expressions.

I'm not even sure you need regex. Once you strip the `file:///` prefix you can use `Path` to turn it into a path and test for "models" or "user" in path.parts. You can get the parts after user with:
```python
user_index = path.parts.index('user')
file = Path(*path.parts[user_index+1:])
```
If you do stick with regexs be sure to match `[\\/]user[\\/]` and `[\\/]models[\\/]` to reduce the chance collision with a filename such as `…/usermode.html`.

You use the same or similar path manipulations elsewhere, so create some helper functions to ease future maintenance.